chia/types/blockchain_format/vdf.py

```diff
@@ -2,39 +2,46 @@
 import traceback
 from dataclasses import dataclass
 from enum import IntEnum
-from typing import Dict, Optional, Tuple
+from typing import Optional
+from functools import lru_cache
 
 from chiavdf import create_discriminant, verify_n_wesolowski
 
 from chia.consensus.constants import ConsensusConstants
 from chia.types.blockchain_format.classgroup import ClassgroupElement
-from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.blockchain_format.sized_bytes import bytes32, bytes100
 from chia.util.ints import uint8, uint64
 from chia.util.streamable import Streamable, streamable
 
 log = logging.getLogger(__name__)
 
-discriminant_cache: Dict[Tuple[bytes32, int], int] = {}
 
+@lru_cache(maxsize=20)
+def get_discriminant(challenge, size_bites) -> int:
```

Comment:
`s/bites/bytes/` :)

Reply:
Types for the arguments? :1st_place_medal:

Reply:
Maybe a bigger size too?

Reply:
heh, I think you wrote this code :)